src/configuration.ts

```diff
@@ -352,10 +458,50 @@ export class ConfigurationManager implements vscode.Disposable
 
     private async addVcpkgToolchain(vcpkgRoot : string)
     {
+        let originToolchain = this.getCMakeVcpkgToolchain();
+        if (originToolchain !== undefined)
+        {
+            if (fs.existsSync(originToolchain))
+            {
+                // check whether the vcpkg path in toolchain is not the same with the path in settings
+                if (this.convertToAbsolutePath(originToolchain) !== this.convertToAbsolutePath(vcpkgRoot + '/scripts/buildsystems/vcpkg.cmake'))
+                {
+                    vscode.window.showErrorMessage('Detected mismatched vcpkg toolchain!');
```

Comment:
See here.
We may also need to do this check before configure cmake.
No idea how to inject vcpkg check in cmake extension.

Reply:
This will be executed when enable vcpkg in extension.

Reply:
> See here. We may also need to do this check before configure cmake. No idea how to inject vcpkg check in cmake extension.

After my exams, I will try to help you out with the development of this extension. If possible, you can prepare a plan and objectives and I can assist you with the work.
I'll be available from 13th December.